pkg/generate/app/pipeline.go

```diff
@@ -341,21 +342,40 @@ func AddServices(objects Objects, firstPortOnly bool) Objects {
 	for _, o := range objects {
 		switch t := o.(type) {
 		case *deploy.DeploymentConfig:
-			ports := UniqueContainerToServicePorts(AllContainerPorts(t.Spec.Template.Spec.Containers...))
-			if len(ports) == 0 {
-				continue
+			svc := addServiceInternal(t.Spec.Template.Spec.Containers, t.ObjectMeta, t.Spec.Selector, firstPortOnly)
+			if svc != nil {
+				svcs = append(svcs, svc)
 			}
-			if firstPortOnly {
-				ports = ports[:1]
+		case *extensions.DaemonSet:
+			// Daemonsets have defaulting logic, if the selector is not set it will default to
+			// the labels on the pod template
+			selector := t.Spec.Template.Labels
+			if t.Spec.Selector != nil {
+				selector = t.Spec.Selector.MatchLabels
```

Comment:
I don't think we default in this direction though.  The daemonset node selector can be destroyed via defaulting, but the pod template spec (the one that the service has to match) is never defaulted, right?